avocado/core/nrunner.py

```diff
@@ -677,6 +677,7 @@ def __init__(self, runnable, identifier=None, status_uris=None,
         self.known_runners = known_runners
         self.spawn_handle = None
         self.output_dir = None
+        self.data = {}
```

Comment:
If this is a metadata, then I would recommend calling it `self.metadata`.

Reply:
Ok, I will update the description